api/controllers/FlairController.js

```diff
@@ -97,9 +97,16 @@ module.exports = {
       var ipAddress = req.headers['x-forwarded-for'] || req.ip;
       // Get IP matches with banned users
       var events_with_ip = await Event.find({content: {contains: ipAddress}, user: {not: req.user.name}});
+
       var matching_alt_usernames = _.uniq(_.map(events_with_ip, 'user'));
-      var matching_banned_users = await User.find({name: matching_alt_usernames, banned: true});
-      var banned_alts = _.map(matching_banned_users, 'name');
+      var matching_users = await User.find({name: matching_alt_usernames});
+      var matching_banned_users = matching_users.filter(user => user.banned);
+      var alt_user_names = _.map(matching_users, 'name');
+
+      var users_with_matching_fcs = await User.find({loggedFriendCodes: flairs.fcs, name: {not: req.user.name}});
+      var logged_fcs = _.flatten(_.map(users_with_matching_fcs, 'loggedFriendCodes'));
+      var matching_friend_codes = _.intersection(flairs.fcs, logged_fcs);
+      var alt_users_fc = _.map(users_with_matching_fcs, 'name');
```

Comment:
Nitpick: This variable name is misleading because it's a list of names, not FCs.